### ISSUE TITLE
API Gateway and Swagger paths

### ISSUE THREAD

**Issue by ashemedai** [awslabs/cfn-python-lint#243]
One problem we ran into was where someone had a template.yaml defining both API Gateway and associated Lambdas. A Lambda specifies something like (truncated to the bare bones):

```yaml
FunctionSomething:
  Type: "AWS::Serverless::Function"
  Properties:
    Events:
      SomeApiName:
        Type: Api
        Properties:
          Path: /{myVar}/{anotherVar}
          RestApiId: !Ref "ServerlessApi"

ServerlessApi:
  Type: "Aws::Serverless::Api"
  Properties:
    DefinitionBody:
      swagger: '2.0'
      paths:
        '/{someVar}/{andAnotherVar}':
          ...
```

This mismatch in paths will lead to logging output like:

```
Sending request to https://lambda.eu-west-1.amazonaws.com/2015-03-31/functions/arn:aws:lambda:eu-west-1:NNNN:function:function-something/invocations
Execution failed due to configuration error: Invalid permissions on Lambda function
Method completed with status: 500
```

And this will lead you down the wrong way of thinking it's permissions in IAM or something, whereas it's something as simple as mismatched identifiers.

@cmmeyer, @kddejong is this something we would consider within the scope of the linter to do something with? I am definitely not suggesting we lint the entire swagger section, that seems solidly outside the scope. But this part seems awfully prone to mistakes, especially as the YAML file grows (which API Gateway makes very easy).

**Comment by cmmeyer** [comment]
Interesting -- we're using the SAM CLI libraries to validate Serverless components. I'm thinking it makes more sense to add as an enhancement on the Serverless side.  I'm reluctant to double up on Serverless validation.  Let me reach out to them.
